litmus-portal/graphql-server/pkg/chaos-workflow/service.go

```diff
@@ -328,35 +328,36 @@ func (c *chaosWorkflowService) ProcessCompletedWorkflowRun(execData ExecutionDat
 	return result, nil
 }
 
-func processWorkflowManifest(workflow *model.ChaosWorkFlowRequest, weights map[string]int) error {
-	var (
-		newWeights       []*model.WeightagesInput
-		workflowManifest v1alpha1.Workflow
-	)
-	err := json.Unmarshal([]byte(workflow.WorkflowManifest), &workflowManifest)
-	if err != nil {
-		return errors.New("failed to unmarshal workflow manifest")
-	}
-
-	if workflowManifest.Labels == nil {
-		workflowManifest.Labels = map[string]string{
-			"workflow_id": *workflow.WorkflowID,
-			"cluster_id":  workflow.ClusterID,
-			"workflows.argoproj.io/controller-instanceid": workflow.ClusterID,
+func updateManifestLabels(labels map[string]string, workflowID string, clusterID string, requiresType bool) map[string]string {
+	if labels == nil {
+		labels = map[string]string{
+			"workflow_id": workflowID,
+			"cluster_id":  clusterID,
+		}
+		if requiresType {
+			labels["type"] = "standalone_workflow"
+		} else {
+			labels["workflows.argoproj.io/controller-instanceid"] = clusterID
 		}
 	} else {
-		workflowManifest.Labels["workflow_id"] = *workflow.WorkflowID
-		workflowManifest.Labels["cluster_id"] = workflow.ClusterID
-		workflowManifest.Labels["workflows.argoproj.io/controller-instanceid"] = workflow.ClusterID
+		labels["workflow_id"] = workflowID
+		labels["cluster_id"] = clusterID
+		if requiresType {
+			labels["type"] = "standalone_workflow"
+		} else {
+			labels["workflows.argoproj.io/controller-instanceid"] = clusterID
+		}
 	}
+	return labels
+}
+
+func generateWeightsFromManifest(templates []v1alpha1.Template, labels map[string]string, weights map[string]int) ([]*model.WeightagesInput, error) {
```

Comment:
Can you please rename this function to generateFaultWeights

Reply:
Okay. Will do it 👍